### ISSUE TITLE
how to adjust the lambda parameter

### ISSUE THREAD

**Issue by KeanuWongves** [LIVIAETS/boundary-loss#53]
Hey, there. I have used ur work with mine, I have some questions about it.
When using it with my task, in the first epochs, the loss seems like really big, I don't how the gradients is, because u know the value of loss is not equivalent to the gradient, ( big loss value may be with small gradient)
so I change the loss function from sum to mean, but it seems like too small maybe, idk. but the metrix like f1-score or iou seems like drop a little, I dont know how to adjust the parameters.
I just use constant lambda for iou and boundary_loss which is 1 and 0.1.
do u have some suggestions for me? thx

**Comment by superbunny38** [comment]
Hello, do you mean the lambda as alpha, which coordinates the ratio between losses?
I set my criterion as 0.01xSoft Dice Loss + 0.99xBoundary Loss + 0.3xAuxiliary Loss, and received the best result in segmentation, in terms of dice coefficient score. 
Feel free to drop by my repo and check out: https://github.com/superbunny38/Multi-Object-Semantic-Segmentation

**Comment by KeanuWongves** [comment]
@superbunny38 
Hey its very kind of u, I will check this out later.
Have a nice day!

**Comment by KeanuWongves** [comment]
@superbunny38  But here is my situation, as long as I use boundary loss the result get drop
and the bigger the bg_loss's weight is, the larger result drops.
I dont know maybe bg_loss is just not suitable for my task

**Comment by superbunny38** [comment]
> @superbunny38 But here is my situation, as long as I use boundary loss the result get drop and the bigger the bg_loss's weight is, the larger result drops. I dont know maybe bg_loss is just not suitable for my task

@q671383789 
Hi :)
I watched the presentation by the authors of this paper on youtube, and they say that if the model's base performance (performance without boundary loss for segmentation in terms of dice coefficient) is not good enough (=unstable), then the boundary loss would not be great. 

I think it might be helpful to share my experience. I was able to receive my best 0.98 dice coefficient on test set in lung segmentation task with the dice loss, but before using boundary loss for fine-tuning the model, my model's performance was already very high (0.96 on test set). However, when I tried using boundary loss for OCT segmentation task, in which I received 0.3 dice score, using boundary loss led to worsening the performance.

I hope that helped :)

**Comment by HKervadec** [comment]
This is weird, I haven't received any notification on my e-mails about this thread (or a few other issues). Hence my delayed response. 

> I set my criterion as 0.01xSoft Dice Loss + 0.99xBoundary Loss + 0.3xAuxiliary Loss, and received the best result in segmentation, in terms of dice coefficient score.
Feel free to drop by my repo and check out: https://github.com/superbunny38/Multi-Object-Semantic-Segmentation

That's very interesting, and good to know, thanks for linking it!

> When using it with my task, in the first epochs, the loss seems like really big

In the case of the boundary loss, the per-pixel gradient is actually the value of the distance map. So you could easily get that information.

> so I change the loss function from sum to mean, but it seems like too small maybe, idk. but the metrix like f1-score or iou seems like drop a little, I dont know how to adjust the parameters.

You could also (as was done in the original paper) adjust the losses weights over time; to use the boundary loss as 'refinement'. That's what the 'scheduler' options are for: https://github.com/LIVIAETS/boundary-loss/blob/master/scheduler.py

But then yes, depending on the task the boundary loss might have no effect, or make things slightly worse (if your results are already very high). Finding the proper balance remains ultimately very task specific, but I am always interested to hear user stories about it.

Hoel
